bson/__init__.py

```diff
@@ -887,26 +897,41 @@ def _dict_to_bson(doc: Any, check_keys: bool, opts: CodecOptions, top_level: boo
 if _USE_C:
     _dict_to_bson = _cbson._dict_to_bson  # noqa: F811
 
+# Inclusive, not inclusive.
+MIN_PYDATETIME = datetime.datetime.min.replace(tzinfo=datetime.timezone.utc).timestamp() * 1000
+MAX_PYDATETIME = datetime.datetime.max.replace(tzinfo=datetime.timezone.utc).timestamp() * 1000 - 1
```

Comment:
Thanks, this all makes sense but it would be better to use _datetime_to_millis here since it already handles with these issues:
```
>>> bson._datetime_to_millis(datetime.datetime.max)
253402300799999
>>> bson._millis_to_datetime(253402300799999, DEFAULT_CODEC_OPTIONS)
datetime.datetime(9999, 12, 31, 23, 59, 59, 999000)
```